modules/history/src/history-helper.js

```diff
@@ -0,0 +1,180 @@
+const SerializableClass = require("abitbol-serializable");
+const self = require("../index.js");
+
+const historyHelper = {
+
+    /**
+     * @param  {Object} currentState
+     * @param  {Object} snapshotState
+     */
+    applySnapshotDifference: (currentState, snapshotState) => { // eslint-disable-line
+        const {
+            dataStore,
+        } = self.app.modules;
+        // We make a flat object list for each object
+        const currentEntities = historyHelper.flattenSnapshot(currentState);
+        const snapshotEntities = historyHelper.flattenSnapshot(snapshotState);
+
+        Object.keys(currentEntities).forEach(
+            (currentEntityKey) => {
+                let entityFound = false;
+                Object.keys(snapshotEntities).forEach(
+                    (snapshotEntityKey) => {
+                        // Check common entity properties
+                        const currentEntity = currentEntities[currentEntityKey];
+                        const snapshotEntity = snapshotEntities[snapshotEntityKey];
+                        if (currentEntityKey === snapshotEntityKey) {
+                            entityFound = true;
+                            const entity = dataStore.getEntity(currentEntityKey);
+                            Object.keys(snapshotEntity).forEach(
+                                (property) => {
+                                    if ((typeof currentEntity[property] !== "undefined"
+                                        && historyHelper.checkDiff(currentEntity[property],
+                                            snapshotEntity[property]))
+                                        || !currentEntity[property]) {
+                                        // Property is different or doesn't exist
+                                        const unserializedEntity = SerializableClass
```

Comment:
You could unserialize the entity only once instead of doing it at each iteration.